extensions/data-transfer/portability-data-transfer-solid/src/main/java/org/datatransferproject/transfer/solid/contacts/SolidContactsImport.java

```diff
@@ -0,0 +1,318 @@
+        strucName.addProperty(VCARD4.family_name, structuredName.getFamily());
+      }
+      if (!Strings.isNullOrEmpty(structuredName.getGiven())) {
+        strucName.addProperty(VCARD4.given_name, structuredName.getGiven());
+      }
+
+      structuredName.getPrefixes()
+          .forEach(prefix -> strucName.addProperty(VCARD4.hasHonorificPrefix, prefix));
+
+      structuredName.getSuffixes()
+          .forEach(suffix -> strucName.addProperty(VCARD4.hasHonorificSuffix, suffix));
+
+      structuredName.getAdditionalNames()
+          .forEach(additional -> strucName.addProperty(VCARD4.hasAdditionalName, additional));
+
+      r.addProperty(VCARD4.hasName, strucName);
+    }
+
+    for (Email email : vcard.getEmails()) {
+      String mailTo = "mailto:" + email.getValue();
+      if (email.getTypes().isEmpty()) {
+        r.addProperty(VCARD4.hasEmail, mailTo);
+      } else {
+        Resource emailResource = personModel.createResource();
+        emailResource.addProperty(VCARD4.value, mailTo);
+        for (EmailType type : email.getTypes()) {
+          for (Resource emailTypeResource :
+              getPhoneOrMailTypes(type.getValue(), MAP_OF_EMAIL_TYPES)) {
+            emailResource.addProperty(RDF.type, emailTypeResource);
+          }
+        }
+        r.addProperty(VCARD4.hasEmail, emailResource);
+      }
+    }
+
+    for (Telephone telephone : vcard.getTelephoneNumbers()) {
+      if (telephone.getTypes().isEmpty()) {
+        r.addProperty(VCARD4.hasTelephone, telephone.getText());
+      } else {
+        Resource telephoneResource = personModel.createResource();
+        telephoneResource.addProperty(VCARD4.value, telephone.getText());
+        for (TelephoneType type : telephone.getTypes()) {
+          for (Resource telTypeResource : getPhoneOrMailTypes(type.getValue(), MAP_OF_PHONE_TYPES))
+            telephoneResource.addProperty(RDF.type, telTypeResource);
+        }
+        r.addProperty(VCARD4.hasTelephone, telephoneResource);
+      }
+    }
+
+    if (vcard.getOrganization() != null) {
+      r.addProperty(VCARD4.organization_name, vcard.getOrganization().getValues().get(0));
+    }
+
+    for (Organization organization : vcard.getOrganizations()) {
+      organization.getValues().stream().forEach(
+          v -> r.addProperty(VCARD4.hasOrganizationName, v));
+    }
+
+    for (Url url : vcard.getUrls()) {
+      r.addProperty(VCARD4.hasURL, url.getValue());
+    }
+
+    for (Note note : vcard.getNotes()) {
+      r.addProperty(VCARD4.hasNote, note.getValue());
+    }
+
+    for (Photo photo : vcard.getPhotos()) {
+      r.addProperty(VCARD4.hasPhoto, photo.getUrl());
+    }
+
+    StringWriter stringWriter = new StringWriter();
+    personModel.write(stringWriter, "TURTLE");
+    return stringWriter.toString();
+  }
+
+  /** Looks up the {@link Resource}s for a given string, that might be comma delimited. **/
+  private static ImmutableList<Resource> getPhoneOrMailTypes(
```

Comment:
final

Reply:
Why mark a private static method final, it has to be final.  (I am agnostic, but IntelliJ complains if you mark them final, which makes me think doing so might be an anti-pattern)